### PR TITLE
Set token in cookies for email verification and google authentificati…

### PR DESCRIPTION
**Pull request by scherbenokk**
…on endpoints

### CODE REVIEW

**Comment by scherbenokk** [comment]
@IharKrasnik updated and checked

File: src/config/routes/index.js [hunk 1]
```diff
@@ -1,17 +1,14 @@
-const jwt = require('koa-jwt');
-
-const config = require('config');
-
-const { urlToken, user } = require('./middlewares');
+const { auth } = require('./middlewares/user.middleware');
```
Comment (IharKrasnik) [inline src/config/routes/index.js]: Only keep the logic to attach user to ctx.state.user (rather null if token is invalid or missing or user object)

Move auth logic to separate middleware (middlewares/auth.middleware)
Throw 401 only if ctx.state.user is not present

File: src/resources/token/token.helper.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const { get } = require('lodash');
```
Comment (IharKrasnik) [inline src/resources/token/token.helper.js]: Get rid of token.helper.

Helper should contain only simple (one-liner like) pieces of code.

Move all methods (getUserIdByToken, removeTokens, createTokens) to accessTokenService

File: src/resources/account/account.helper.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const psl = require('psl');
```
Comment (IharKrasnik) [inline src/resources/account/account.helper.js]: Get rid of account.helper

Add cookie.helper instead, with methods responsible only to adding tokens to cookies (without database access)
#setTokenCookies({ accessToken, refreshToken })
#unsetTokenCookies()


To be able to reuse setTokens/unsetTokens method put them in auth.service

File: src/app.constants.js [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+    TOKENS: 'tokens',
   },
+  TOKEN_TYPES: {
+    REFRESH: 'refresh',
+    ACCESS: 'access',
+  },
+  COOKIES: {
+    ACCESS_TOKEN: 'access_token',
+    REFRESH_TOKEN: 'refresh_token',
+    EXPIRED_AT: 'expired_at',
```
Comment (IharKrasnik) [inline src/app.constants.js]: Rename EXPIRED_AT to EXPIRE_AT

File: src/migrations/migrations/1.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
+const db = require('db');
 const Migration = require('migrations/migration');
 
-const migration = new Migration(1, 'Example');
+const migration = new Migration(1, 'Set ttl index on tokens collection');
 
 migration.migrate = async () => {
+  await db.collection('tokens').createIndex({ expireAt: 1 }, { expireAfterSeconds: 0 });
```
Comment (IharKrasnik) [inline src/migrations/migrations/1.js]: Remove from migrations.
Put this code inside tokenService itself (createIndex aka ensureIndex is idempotent)

File: src/services/socket.io.service.js [hunk 6]
```diff
@@ -1,22 +1,20 @@
 const io = require('socket.io')();
```
Comment (IharKrasnik) [inline src/services/socket.io.service.js]: socketIo.service.js was a correct naming (there's no such entity as 'io' in project, it's part of the library name), please rename back

File: src/resources/account/account.helper.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    domain: cookiesDomain,
+  });
+
+  ctx.cookies.set(COOKIES.REFRESH_TOKEN, refreshTokenEntity.value, {
+    httpOnly: true,
+    expires: refreshTokenEntity.expireAt,
+    domain: cookiesDomain,
+  });
+
+  ctx.cookies.set(COOKIES.EXPIRED_AT, accessTokenEntity.expireAt, {
```
Comment (IharKrasnik) [inline src/resources/account/account.helper.js]: Do we need a separate cookie for expire_at? It's already stored in access token cookie

File: src/config/routes/middlewares/validateSessionDuration.middleware.js [hunk 8]
```diff
@@ -1,24 +1,17 @@
...
-    await next();
-    return;
-  }
-
-  const user = await userService.findById(ctx.state.user._id);
-  ctx.assert(user, 'Not authorized: user not found', 401);
+const validateSessionDuration = async (ctx, next) => {
+  const { user } = ctx.state;
 
   await userService.updateLastRequest(ctx.state.user._id);
```
Comment (IharKrasnik) [inline src/config/routes/middlewares/validateSessionDuration.middleware.js]: Please use `user`instead of `ctx.state.user` here for consistency

**Pull request merged**